tsdb/notify.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tsdb
+
+const (
+	WALCorruptionDetectedText    = "WAL corruption has been detected."
+	WALCorruptionDetectedDetails = "Review the logs for more details. Ensure disk integrity and file system health. Consider clearing the WAL directory and restarting Prometheus."
+)
+
+// NotificationService defines the interface for adding and deleting
+// notifications.
+type NotificationService interface {
```

Comment:
While often it's best to define interfaces at the usage site, if it's an interface that isn't really specific to the user at (`tsdb`) at all and will likely be used by others as well in the future, I'd define it (along with the nop implementation) in the same package as the actual notifier code.

Reply:
Hm yeah, that's a valid consideration indeed. Then maybe since the TSDB is the only user for now, let's change it later when it's worth it. If we don't want to depend on anything under `web/...` at all from the TSDB, then we could still move the notifier code to a separate hierarchy then (although I still think `model` wouldn't be quite the right place for it).